pkg/controller/provider/container/vsphere/collector.go

```diff
@@ -697,41 +697,53 @@ func (r *Collector) propertySpec() []types.PropertySpec {
 			},
 		},
 		{ // VM
-			Type: VirtualMachine,
-			PathSet: []string{
-				fName,
-				fParent,
-				fUUID,
-				fFirmware,
-				fFtInfo,
-				fCpuAffinity,
-				fCpuHotAddEnabled,
-				fCpuHotRemoveEnabled,
-				fMemoryHotAddEnabled,
-				fNumCpu,
-				fNumCoresPerSocket,
-				fMemorySize,
-				fDevices,
-				fExtraConfig,
-				fGuestName,
-				fGuestID,
-				//fTpmPresent,
-				fBalloonedMemory,
-				fVmIpAddress,
-				fStorageUsed,
-				fDatastore,
-				fNetwork,
-				fRuntimeHost,
-				fPowerState,
-				fConnectionState,
-				fIsTemplate,
-				fSnapshot,
-				fChangeTracking,
-			},
+			Type:    VirtualMachine,
+			PathSet: r.vmPathSet(),
 		},
 	}
 }
 
+func (r *Collector) vmPathSet() []string {
+	apiVer := strings.Split(r.client.ServiceContent.About.ApiVersion, ".")
+	majorVal, _ := strconv.Atoi(apiVer[0])
+	minorVal, _ := strconv.Atoi(apiVer[1])
+
+	pathSet := []string{
+		fName,
+		fParent,
+		fUUID,
+		fFirmware,
+		fFtInfo,
+		fCpuAffinity,
+		fCpuHotAddEnabled,
+		fCpuHotRemoveEnabled,
+		fMemoryHotAddEnabled,
+		fNumCpu,
+		fNumCoresPerSocket,
+		fMemorySize,
+		fDevices,
+		fExtraConfig,
+		fGuestName,
+		fGuestID,
+		fBalloonedMemory,
+		fVmIpAddress,
+		fStorageUsed,
+		fDatastore,
+		fNetwork,
+		fRuntimeHost,
+		fPowerState,
+		fConnectionState,
+		fIsTemplate,
+		fSnapshot,
+		fChangeTracking,
+	}
+
+	if majorVal >= 6 && minorVal >= 7 {
```

Comment:
that's not good since it would fail on things like 7.0
how about adding a unit test for this?